### PR TITLE
Updating how Test-Proxy shells out, adding tests for long running add operations

### PR DESCRIPTION
**Pull request by scbedd**
- Bunch of small files
- Couple of large files

This is an attempt at repro-ing the issue #4286

I've got a working solution that I really like the look of. We shouldn't ever get in a deadlock situation now, and the LargePushPerf tests are passing locally for the worst case scenario (1000 files of ~150KB random bytes each)

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationPushTests.cs [hunk 1]
```diff
@@ -363,5 +361,81 @@ public async Task Scenario4(string inputJson)
...
+        /// 5. Verify assets.json was updated with the new commit Tag
+        /// </summary>
+        /// <param name="inputJson"></param>
+        /// <returns></returns>
+        [EnvironmentConditionalSkipTheory]
+        [InlineData(1000, 0.15)] // 1000 0.15 MB files
+        [InlineData(150, 0.25)]  // 150  0.25 MB files
+        [InlineData(100, 0.50)]  // 100  0.5 MB files
+        [InlineData(10, 1.50)]   // 10   1.5 MB files
+        [InlineData(1, 1024)]    // 11   1 GB file
```
Comment (JimSuplizio) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationPushTests.cs]: You can skip the GB file, git will complain if you try and push anything over 100 MB.

**Review by JimSuplizio — APPROVED** [review]
One minor suggestion to the test, otherwise things look good. I'd pulled the changes locally and run them against java's sdk\storage which has 4693 files totaling 666MB. It took 7 minutes for the initial push, but subsequent pushes with small numbers of updated files were less than 10 seconds.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationPushTests.cs (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/IntegrationTests/GitStoreIntegrationPushTests.cs]: You're absolutely right and we removed it.

**Pull request merged**